### ISSUE TITLE
[oracle] LT01 should not affect %type and %rowtype attributes

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8070]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The example below is flagged with LT01 while being linted.
See: https://docs.oracle.com/en/database/oracle/oracle-database/26/lnpls/TYPE-attribute.html

### Expected Behaviour

% should not be considered as a binary operator in %type/%rowtype context.

### Observed Behaviour

`Expected single whitespace between naked identifier and binary operator '%'.`
`Expected single whitespace between binary operator '%' and 'type' keyword.`
`Expected single whitespace between binary operator '%' and 'rowtype' keyword.`

### How to reproduce

Linting the following example will flag the %type and %rowtype attributes:
```sql
function f_ex(
    p_ex1 in ex_tab.a%type,
    p_ex2 in ex_tab%rowtype
) return boolean;
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by apoorva-01** [comment]
Poked at this a bit. The `%` in `%type`/`%rowtype` is getting read as the modulo operator, and LT01 wants spaces around operators, hence the `a % type` complaint. It's everywhere `%` shows up too, not just here, cursor attrs like `c1%notfound`, var decls like `v emp%rowtype`, same deal.

There's already a nice way to handle symbols that shouldn't be spaced (`::`, `@` and friends each get their own type + a `touch` layout entry). Figured I'd do the same for the attribute `%` so it stops being treated as modulo. Keeps real `a % b` working.

Before I build it, which way do you want to go, own type + layout entry, or just special-case it in LT01? And should I cover all the `%` spots or just `%type`/`%rowtype`?

Happy to send a PR once you point me the right way.

**Comment by osarier** [comment]
Hi @apoorva-01 
thanks for looking into this issue!

First of all I realized that my %rowtype example was wrong. I fixed that now.

I missed the other cases like cursor attrs and var decls that you mentioned while creating this issue but I think it makes sense to have a solution that takes care of these as well.

Since other similiar issues were resolved by introducing a new type and a corresponding layout entry I think this solution makes sense here as well.
